src/openrct2/Game.cpp

```diff
@@ -1384,7 +1384,8 @@ void game_autosave()
         timeName, sizeof(timeName), "autosave_%04u-%02u-%02u_%02u-%02u-%02u%s", currentDate.year, currentDate.month,
         currentDate.day, currentTime.hour, currentTime.minute, currentTime.second, fileExtension);
 
-    limit_autosave_count(NUMBER_OF_AUTOSAVES_TO_KEEP, (gScreenFlags & SCREEN_FLAGS_EDITOR));
+    int32_t autosavesToKeep = gConfigGeneral.autosave_amount;
+    limit_autosave_count(autosavesToKeep - 1, (gScreenFlags & SCREEN_FLAGS_EDITOR));
```

Comment:
Since this code makes it unnecessary, you should remove `NUMBER_OF_AUTOSAVES_TO_KEEP`.